aiter/ops/triton/hstu_attention.py

```diff
@@ -0,0 +1,1213 @@
+            )
+    for start_m in tl.range(low, high, BLOCK_M, loop_unroll_factor=UNROLL):
+        start_m = tl.multiple_of(start_m, BLOCK_M)
+        dk, dv = _hstu_attn_bwd_one_block(
+            start_m=start_m,
+            offs_n=offs_n,
+            offs_m=offs_m,
+            q_ptrs_trans=q_ptrs_trans,
+            dq_ptrs_trans=dq_ptrs_trans,
+            mask_n=mask_n,
+            do_ptrs=do_ptrs,
+            dk=dk,
+            dv=dv,
+            k=k,
+            v=v,
+            pos_offs_n=pos_offs_n,
+            seq_len=seq_len,
+            n_targets=n_targets,
+            max_ids=max_ids,
+            contextual_seq_len=contextual_seq_len,
+            max_attn_len=max_attn_len,
+            LOCK=LOCK,
+            stride_qm=stride_qm,
+            stride_dom=stride_dom,
+            stride_dqm=stride_dqm,
+            alpha=alpha,
+            MAX_SEQ_LEN=MAX_SEQ_LEN,
+            CAUSAL=CAUSAL,
+            HAS_MULTIPLE_TARGETS=HAS_MULTIPLE_TARGETS,
+            HAS_CONTEXTUAL_SEQ_LEN=HAS_CONTEXTUAL_SEQ_LEN,
+            HAS_MAX_ATTN_LEN=HAS_MAX_ATTN_LEN,
+            ALLOW_TF32=ALLOW_TF32,
+            BLOCK_M=BLOCK_M,
+            BLOCK_N=BLOCK_N,
+            ATOMIC_ADD=ATOMIC_ADD,
+        )
+    # write-back
+    dv_ptrs = DV + (offs_n[:, None] * stride_dvn + offs_v_d[None, :])
+    dk_ptrs = DK + (offs_n[:, None] * stride_dkn + offs_qk_d[None, :])
+    dk = dk * alpha
+    tl.store(dv_ptrs, dv.to(k.dtype), mask=mask_n[:, None])
+    tl.store(dk_ptrs, dk.to(k.dtype), mask=mask_n[:, None])
+
+
+def _bwd_pre_hook(nargs):
+    nargs["DQ"].zero_()
+    if nargs["SEQUENCE_PARALLEL"] is True:
+        nargs["LOCK"].zero_()
+
+
+def _get_bw_configs() -> List[triton.Config]:
+    configs = []
+    for BLOCK_M in [32, 64]:
+        for BLOCK_N in [32, 64]:
+            for num_stages in [1, 2]:
+                for num_warps in [4, 8]:
+                    for matrix_instr_nonkdim in [16, 32]:
+                        for waves_per_eu in [0, 2, 4]:
+                            for sp in [True, False]:
+                                configs.append(
+                                    triton.Config(
+                                        {
+                                            "BLOCK_M": BLOCK_M,
+                                            "BLOCK_N": BLOCK_N,
+                                            "matrix_instr_nonkdim": matrix_instr_nonkdim,
+                                            "waves_per_eu": waves_per_eu,
+                                            "SEQUENCE_PARALLEL": sp,
+                                            "UNROLL": 1,
+                                        },
+                                        num_stages=num_stages,
+                                        num_warps=num_warps,
+                                        pre_hook=_bwd_pre_hook,
+                                    )
+                                )
+    return configs
+
+
+@triton.autotune(
```

Comment:
We don't use triton autotuner in AITER. This is bad during inference for latency
For training, I am working on having the autotuner turn on optionally, but by default we can't have autotuner in AITER